elasticsearch.md

```diff
@@ -1719,64 +1718,134 @@ readonlyrest:
+One of the neatest features in ReadonlyREST is that you can use dynamic variables inside most values of the following rules: `indices`, `users`, `groups`, `groups_and`, `fields`, `filter`,  `repositories`, `snapshots`, `response_fields`, `uri_re`, `x_forwarded_for`, `hosts_local`, `hosts`, `kibana_index`, `kibana_template_index`. The variables are related to different contexts:
+* `acl` - the context of data collected in authentication and authorization rules of the current block:
+    * `@{acl:user}` gets replaced with the username of the successfully authenticated user. Using this variable is allowed only in blocks where one of the rules is an authentication rule of course it must be rule different from the one containing the given variable.
+    * `@{acl:current_group}` gets replaced with user's current group. Usually resolved by authorization rule defined in a block, but value can be also retrieved by means of kibana plugin. This variable doesn't specify usage requirements.
+    * `@{acl:available_groups}` gets replaced with available groups found in the authorization rule (bacause group strings will be surrounded with double quotes and joined with a comma)
+* `header` - the context of ES HTTP request headers
+    * `@{header:<header_name>}` gets replaced with the value of the HTTP header with name `<header_name>` included in the incoming request \(useful when reverse proxies handle authentication\)
+* `jwt` - the context of JWT passed in the Authorization header 
+    * `@{jwt:<json_path>}` get replaced with value (or values) found in JWT under the given JSON path
+
+#### Dynaminc variables exploding
 
-* `@{acl:user}` gets replaced with the username of the successfully authenticated user. Using this variable is allowed only in blocks where one of the rules is authentication rule \(of course it must be rule different from the one containing given variable\).
-* `@{user}` old style user variable definition. Preferred approach is to use `@{acl:user}`.
-* `@{acl:current_group}` gets replaced with user's current group. Usually resolved by authorization rule defined in block, but value can be also retrieved by means of kibana plugin. This variable doesn't specify usage requirements.
-* `@{xyz}` gets replaced with any `xyz` HTTP header included in the incoming request \(useful when reverse proxies handle authentication\)
+A value resolved from a dynamic variable is a string. Some rules, like `indices` one, have multivalue context (you can configure serveral indices names in it). 
 
-#### Indices from user name
+Let's assume we have a request with the header: `APPS: app1,app2,app3`. Doing something like that: 
+
+```yaml
+indices: ["logstash_@{header:apps}"]
+```
+
+We should expect it to be resolved to:
+
+```yaml
+indices: ["logstash_app1,app2,app3"]
+```
+
+for this particular request. I don't think it would be helpful.
+But there is an `explode` attribute for dynamic variables. Doing:
```

Comment:
I'd say explode is more of a function than an attribute, right?

Reply:
hmm, yes, from the user perspective it's more like a function. 
From the env perspective, it's a marker (or attribute) that changes the way how it's resolved. But the first one seems to be more intuitive.

Reply:
In general we should struggle to keep up the blatantly artificial illusion of simplicity, hiding the implementation complexity and handling obvious corner cases and fallbacks under the hood. So the user's mental model stays extra simple. This gives the "it just works" feeling.  

Imagine the difference between setting up "suspend to disk" in Linux (2 hours googling, learn CPU states, swap partitions,...),  and just closing the lid on your Macbook and go by your day. 

MacOS handles all the same things, but they are hidden behind the fluffy curtains of automation and great UX.

Similarly for our case:

DOCUMENTATION=Close the lid, go by your day.
IMPLEMENTATION=Screen brightness: 0%; Suspend to memory for 40 minutes; if still idle after 40m:  if partition does not exist, create a swap file; dump RAM to swap; try set CPU state to S4, if failed, try S3.

Never leak complexity in documentation :)